pkg/publisher/util/utils.go

```diff
@@ -23,35 +23,63 @@ func NewPublisherProvider(
 	ctx context.Context,
 	storagePath string,
 	cm *system.CleanupManager,
-	ipfsConnect string,
-	localConfig *types.LocalPublisherConfig,
+	cfg types.PublishersConfig,
+	defaultLocalPublisher types.LocalPublisher,
 ) (publisher.PublisherProvider, error) {
-	noopPublisher := noop.NewNoopPublisher()
-	s3Publisher, err := configureS3Publisher(storagePath, cm)
-	if err != nil {
-		return nil, err
-	}
+	providers := make(map[string]publisher.Publisher)
 
-	localPublisher := local.NewLocalPublisher(ctx, localConfig.Directory, localConfig.Address, localConfig.Port)
-
-	providers := map[string]publisher.Publisher{
-		models.PublisherNoop:  tracing.Wrap(noopPublisher),
-		models.PublisherS3:    tracing.Wrap(s3Publisher),
-		models.PublisherLocal: tracing.Wrap(localPublisher),
+	if cfg.Enabled(models.PublisherNoop) {
+		providers[models.PublisherNoop] = noop.NewNoopPublisher()
 	}
 
-	if ipfsConnect != "" {
-		ipfsClient, err := ipfs_client.NewClient(ctx, ipfsConnect)
+	if cfg.Enabled(models.PublisherS3) {
+		s3Publisher, err := configureS3Publisher(storagePath, cm)
 		if err != nil {
 			return nil, err
 		}
-		ipfsPublisher, err := ipfs.NewIPFSPublisher(ctx, *ipfsClient)
+		providers[models.PublisherS3] = tracing.Wrap(s3Publisher)
+	}
+
+	if cfg.Enabled(models.PublisherLocal) {
+		var (
+			localPublisher *local.Publisher
+			err            error
+		)
+		if cfg.Local.Installed() {
+			localPublisher, err = local.NewLocalPublisher(
+				ctx,
+				cfg.Local.Directory,
+				cfg.Local.Address,
+				cfg.Local.Port,
+			)
+		} else {
+			localPublisher, err = local.NewLocalPublisher(
+				ctx,
+				defaultLocalPublisher.Directory,
+				defaultLocalPublisher.Address,
+				defaultLocalPublisher.Port,
+			)
+		}
```

Comment:
This publisher is tricky to configure since:
1. Some test modify the config directly to configure it
2. Others modify the node config (not the new config struct) to enabled it (like devstack tests, which modifies the port to a random open port)
3. Its the default publisher, and therefor must always be enabled and "installed", in the event there isn't a config value for it provided by the user, we use a default setting from `pkg/node/config_compute.go`